site-builder/src/site/manager.rs

```diff
@@ -142,6 +146,87 @@ impl SiteManager {
         Ok(())
     }
 
+    /// Handles the storage of existing resources on Walrus,
+    /// checking their epochs and storing if necessary.
+    async fn update_existing_resources(&self, existing_resources: &ResourceSet) -> Result<()> {
+        for resource in &existing_resources.inner {
+            let blob_id = &resource.info.blob_id;
+
+            tracing::debug!(
+                resource=?resource.full_path,
+                blob_id=%blob_id,
+                unencoded_size=%resource.unencoded_size,
+                "Checking blob on Walrus"
+            );
+
+            // Always attempt to store the resource on Walrus
+            display::action(format!(
+                "Storing existing resource on Walrus: {}",
+                &resource.info.path
+            ));
+
+            // Store the resource on Walrus and check its status
+            let store_output =
+                self.walrus
+                    .store(resource.full_path.clone(), self.epochs, self.force)?;
+            display::done();
+
+            // Handle the result of the store operation
+            match store_output.0 {
+                BlobStoreResult::AlreadyCertified { end_epoch, .. } => {
+                    if end_epoch < self.epochs {
+                        tracing::info!(
+                            "Resource {} is certified for fewer epochs ({} < {}),
+                            storing again with --force",
+                            resource.info.path,
+                            end_epoch,
+                            self.epochs
+                        );
+
+                        display::action(format!(
+                            "Re-storing resource on Walrus with --force: {}",
+                            &resource.info.path
+                        ));
+                        let _force_output =
+                            self.walrus
+                                .store(resource.full_path.clone(), self.epochs, true)?;
+                        display::done();
+                    } else {
+                        tracing::info!(
+                            "Skipping resource {} as it's already certified
+                            for the required epochs ({} >= {})",
+                            resource.info.path,
+                            end_epoch,
+                            self.epochs
+                        );
+                    }
+                }
+                BlobStoreResult::NewlyCreated { .. } => {
+                    tracing::info!(
+                        "Resource {} was newly created on Walrus",
+                        resource.info.path
+                    );
+                }
+                BlobStoreResult::MarkedInvalid { .. } => {
```

Comment:
Important: if a blob is marked invalid, we can never upload it successfully. This arm should just error out.

Reply:
The arm has been removed!